portal-impl/src/com/liferay/portlet/dynamicdatamapping/service/impl/DDMTemplateLocalServiceImpl.java

```diff
@@ -1409,14 +1412,13 @@ protected String formatScript(String type, String language, String script)
 
 			try {
 				script = DDMXMLUtil.validateXML(script);
-				script = DDMXMLUtil.formatXML(script);
 			}
-			catch (Exception e) {
-				throw new TemplateScriptException();
+			catch (PortalException pe) {
+				throw new TemplateScriptException(pe);
 			}
 		}
 
-		return script;
+		return DDMXMLUtil.formatXML(script);
```

Comment:
I think that this should still be in a try/catch and this is also changing the flow of logic (now it always formats it instead of when the condition is true). Why did you move it out?